### ISSUE TITLE
Proposal to update project code structure

### ISSUE THREAD

**Issue by sickelap** [LibrePhotos/librephotos#709]
In order to make the project easier to navigate and reason about the features it implements, I would like to propose an improvement to the backend code structure. Code structure would follow “better” practices, and it would be familiar to python (Django) developers. Another improvement that could be done is to make views “thinner” by moving logic either to models or a service.

There are many ways how code could be structured. It mainly boils down to personal preference. Following structure is one of many ways how this could be done.

Please feel free to share your preferences.

```
librephotos/
- static/
- librephotos/
  - migrations/
  - __init__.py
  - settings.py (or a folder with settings for different envs)
  - urls.py (imports urls from apps)
  - wsgi.py
- apps/
  - api/ (base API implementation; i.e. API docs, Silk, etc.)
    - __init__.py
    - apps.py (meta)
    - urls.py (base endpoints definition)
    - views.py (optional?)
    - ...
  - user/ (user specific features)
    - __init__.py
    - apps.py (meta)
    - admin.py (if module requires admin interface; optional)
    - models.py (or a folder with models in a separate files)
    - urls.py (if module has enough endpoints to justify it; optional)
    - views.py
    - tests.py
    - ...
  - photos/ (photos specific features)
  - nextcloud/
  - ...
manage.py
requirements*.txt
...
```

**Comment by derneuere** [comment]
Yes, this was what I was going for. I entangled all the different models, views and a serializers to be its own files. The next step would be to have a folder for every model. 

A couple of things that are important to me:
- There should also be a migrations folder and a serializer.py per model
- How can we do this iteratively?

Yes, the views can definitely be thinner, but we just have to keep in mind to not change the API.

**Comment by sickelap** [comment]
Of course, the goal is not only to make code more manageable, but also to keep existing functionality.

For the time being, I think all migrations could go into the main folder, i.e., `librephotos`. All new migrations could be done in respective module/app. If there will be a good reason to move existing ones into modules/apps it always can be done later.

What do you mean by “interactively” ?

**Comment by sickelap** [comment]
Ah, misread “iteratively” :rofl: 

Initially, I think we can move `nextcloud` and `api` to apps/. Then we can start splitting `api` to modules.

Because folder structure will be changing and Django resolve table names by module/app path, the default table name “resolution” will break. We will have to add meta to each model to specify table name explicitly.

**Comment by derneuere** [comment]
Yes, that makes sense 👍 I will open up a PR with my big files refactor, and then I think we can start migrating to the better code structure
